docs(README) update automation guide

kikito opened the pull request.

hishamhm commented on README.md:
```diff
@@ -75,30 +75,39 @@ The Enterprise documentation uses paid algolia indices, which auto-update every
+  - On the kong/docs repo, run `KONG_VERSION=0.14.x luajit ./autodoc-nav/run.lua` (replace `0.14.x` with current release). This will merge the navigation
+    info in the right place.
+  - These commands generate a big file for the admin API and a smaller file for the navigation, which needs to be inserted in a
+  - specific place in the navigation yaml file.
```
@kikito The smaller file is used as input by the `autodoc-nav` script. And it also generates another big Admin API file specific for DB-less.

hishamhm commented on README.md:
```diff
@@ -75,30 +75,39 @@ The Enterprise documentation uses paid algolia indices, which auto-update every
     * The command will completely overwrite the file, including text before and after the list of vars.
     * The data used for the before/after parts can be found in `autodoc-conf/data.lua`
 
+- Once everything is generated, git add, git push, and review.
```
```suggestion
- Once everything is generated, submit the generated changes as a pull request.
```

I think it's better to either list the actual commands exactly as need to be executed, or assume the reader knows how to open a pull request (this sits halfway: "git add, git push" look like commands to run but are shorthands that assume that the reader knows how to open a branch, push to that branch, etc)

hishamhm commented on README.md:
```diff
@@ -75,30 +75,39 @@ The Enterprise documentation uses paid algolia indices, which auto-update every
     * Generate a new `cli.md` in the path corresponding to the provided KONG_VERSION.
 - To generate the Configuration docs:
-  - Run: `KONG_PATH=path/to/your/kong/folder KONG_VERSION=0.14.x gulp conf-docs`
+  - In kong/docs: `KONG_PATH=path/to/your/kong/folder KONG_VERSION=0.14.x gulp conf-docs` (replace 0.14.x with current release)
```
Not something to change here but a heads-up: none of the automation that is currently in place for running the generation scripts at release time use `gulp` and the associated JavaScript plumbing. If any changes are made to the process using `gulp`, it won't be picked up there.

Ideally, what should happen is that a "docs generation script" should be extracted out of the "Kong release script" and used both there and here, which would turn the whole process into a one-liner.

kikito commented:
Thanks for the feedback, I made the following changes to reflect hisham's feedback:
* Added a note about the Core team using a different method
* Added a link to our script
* Removed the references to gulp when alternatives existed.
* Added a note in the PDK section pointing out that we don't use gulp for that task.
